app/controllers/auth/signUp.js

```diff
@@ -0,0 +1,38 @@
+const { User } = require('../../models');
+const { checkWoloxEmail, validateEmailExists, validatePassword } = require('../../../utils/validations');
+const hashPassword = require('../../../utils/hashPassword');
+const { NOT_BELONG_COMPANY, EMAIL_ALREADY_EXISTS, INVALID_PASSWORD } = require('../../errors');
+const logger = require('../../logger');
+
+exports.signUp = async (req, res) => {
+  const { name, lastName, email, password } = req.body;
+  try {
+    const isEmailValid = checkWoloxEmail(email);
+    if (!isEmailValid) {
+      return res.status(400).json({ message: NOT_BELONG_COMPANY });
+    }
+
+    const emailExists = await validateEmailExists(email);
+    if (emailExists) {
+      return res.status(422).json({ message: EMAIL_ALREADY_EXISTS });
```

Comment:
Throw an error here. And this http code is not the right one for this situation, use `409` instead

Reply:
Done